### ISSUE TITLE
Exit call in Fortran tests causes random exit codes with one version of the XLF compiler

### ISSUE THREAD

**Issue by colleeneb** [SOLLVE/sollve_vv#168]
Mentioning an issue I ran into with an XL compiler [1] for the record.

In ompvv.F90, there is a use of "exit". Exit is a gfortran extension, not part of fortran standard and not always portable. E.g, this apparently doesn't work properly on some version of XL Fortran [1], like the one I have :). But you guys run on Summit and it's ok there, though?

Basically when I run the Fortran tests with the XL compiler [1], I get incorrect fail/pass rates, since the "exit" call is returning random exit codes. An example is below. note it passes on the device but has totally random exit codes. If I rerun, I get different exit codes for the test, too.

```
 running: bin/declare_target_module.F90.FOR.run 
/home/bertoni/tmp/xl/sollve_vv/sys/scripts/run_test.sh bin/declare_target_module.F90.o 1 2>&1 | tee -a logs/declare_target_module.F90.log && echo "PASS" > _ompvv_temp_result_.exitstatus.1307 || echo "FAIL" > _ompvv_temp_result_.exitstatus.1307
declare_target_module.F90.o: FAIL. exit code: 200
declare_target_module.F90.o:                                                                                                                                                                                                                                        
[OMPVV_INFOMSG declare_target_module.F90:40] Test is running on device                                                                                                                                                                                              
[OMPVV_INFOMSG ompvv.F90:251] Test ran with no errors                                                                                                                                                                                                               
[OMPVV_RESULT declare_target_module.F90] Test passed on the device.
``` 

Because of this, I'd suggest replacing `#define OMPVV_REPORT_AND_RETURN()` in `ompvv.F90` with portable 
```
#define OMPVV_REPORT_AND_RETURN() if( report_and_set_errors(__FILENAME__) .ne. 0) error stop EXIT_FAILURE
```
Or if you only call exit() when there's an error, then you can avoid this issue, too.
```
#define OMPVV_REPORT_AND_RETURN() if( report_and_set_errors(__FILENAME__) .ne. 0) call exit(EXIT_FAILURE)
```

However, I haven't had this issue with any other compiler, so maybe it's not worth changing. I still wanted to report it just in case, though.

[1]

```
$ xlf -qversion
IBM XL Fortran for Linux, V16.1.1 (Community Edition)
Version: 16.01.0001.0003
```

**Comment by colleeneb** [comment]
To try to be super clear, this is the issue:
```
[bertoni@login3.summit summit]$ cat t.F90 
program test
implicit none
call exit(10)
end program test

[bertoni@login3.summit summit]$ xlf t.F90 
** test   === End of Compilation 1 ===
1501-510  Compilation successful for file t.F90.

[bertoni@login3.summit summit]$ ./a.out 
[bertoni@login3.summit summit]$ echo $?
224

[bertoni@login3.summit summit]$ ./a.out 
[bertoni@login3.summit summit]$ echo $?
240

[bertoni@login3.summit summit]$ ./a.out 
[bertoni@login3.summit summit]$ echo $?
32

```
Calling exit results in random exit codes, not what is passed in. `exit` is a GNU extension, so it looks like IBM compiler does not support it.

**Comment by spophale** [comment]
Thanks @colleeneb ! I agree that we should not use features that are not standard. Once we have a fix we will add it to the issue.

**Comment by tob2** [comment]
Regarding the use of `stop` and `error stop`, the Fortran standard (here: 2018) has:

* “When an image is terminated by a STOP or ERROR STOP statement, its stop code, if any, is made available in a processor-dependent manner. If the stop-code is an integer, it is recommended that the value be used as the process exit status, if the processor supports that concept”
* “If the stop-code in a STOP statement is of type character or does not appear, or if an end-program-stmt is executed, it is recommended that the value zero be supplied as the process exit status, if the processor supports that concept.”
* “If the stop-code in an ERROR STOP statement is of type character or does not appear, it is recommended that a processor-dependent nonzero value be supplied as the process exit status, if the processor supports that concept.”

Thus, I would simply use `stop <number>` and check for the exist status. Or `stop "passed"` / `error stop "failed"` or …

**Comment by colleeneb** [comment]
This was driving me crazy (I wanted to do comparative tests), so I submitted PR to fix it #175 . If it isn't what people wanted, let me know.
